sse2neon.h

```diff
@@ -1242,22 +1242,30 @@ FORCE_INLINE __m128i _mm_shuffle_epi32_default(__m128i a,
 
 #define _mm_blend_epi16(a, b, imm)                       \
     ({                                                   \
-        static const uint16_t _mask[8] = {               \
-            ((imm) & 0x1) * 0xFFFF,                      \
-            (((imm) >> 1) & 0x1) * 0xFFFF,               \
-            (((imm) >> 2) & 0x1) * 0xFFFF,               \
-            (((imm) >> 3) & 0x1) * 0xFFFF,               \
-            (((imm) >> 4) & 0x1) * 0xFFFF,               \
-            (((imm) >> 5) & 0x1) * 0xFFFF,               \
-            (((imm) >> 6) & 0x1) * 0xFFFF,               \
-            (((imm) >> 7) & 0x1) * 0xFFFF                \
+        const uint16_t _mask[8] = {                      \
+            ((imm) & 1) ? 0xFFFF : 0x0000,               \
+            ((imm) & 2) ? 0xFFFF : 0x0000,               \
+            ((imm) & 4) ? 0xFFFF : 0x0000,               \
+            ((imm) & 8) ? 0xFFFF : 0x0000,               \
+            ((imm) & 16) ? 0xFFFF : 0x0000,              \
+            ((imm) & 32) ? 0xFFFF : 0x0000,              \
+            ((imm) & 64) ? 0xFFFF : 0x0000,              \
+            ((imm) & 128) ? 0xFFFF : 0x0000,             \
         };                                               \
         uint16x8_t _mask_vec = vld1q_u16(_mask);         \
-        uint16x8_t _a = vreinterpretq_m128i_u16(a);      \
-        uint16x8_t _b = vreinterpretq_m128i_u16(b);      \
-        vreinterpretq_u16_m128i(vbslq_u16(_b, _a, _mask_vec)); \
+        uint16x8_t _a = vreinterpretq_u16_m128i(a);      \
+        uint16x8_t _b = vreinterpretq_u16_m128i(b);      \
+        vreinterpretq_m128i_u16(vbslq_u16(_mask_vec, _b, _a)); \
     })
```

Comment:
I think the fixing should be directly applied on the commit 46b27ccc3feefab482430a944a5de11afd96393a.

Reply:
Huh? I don't know what you mean.

Reply:
I make it more clear in the comment https://github.com/DLTcollab/sse2neon/pull/16#issuecomment-577957540.